File: problem010.cpp
#include "problem010.h"
#include <iostream>
using namespace std;

void problem010(int upperlimit = 2000000) {
    int primes[upperlimit];
    long long i, j;
    for (i = 0; i < upperlimit; i++) {
        primes[i] = 1;
    }
    primes[0] = 0;
    primes[1] = 0;
    for (i = 2; i < upperlimit; i++) {
        if (primes[i] == 1) {
            for (j = i * i; j < upperlimit ; j += i) {
                primes[j] = 0;
            }
        }
    }
    j = 0;
    for (i = 0; i < upperlimit; i++) {
        if (primes[i] == 1) {
            j += i;
        }
    }
    cout << "010: " << j << endl;
}

File: py/problem010.py
"""The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

Find the sum of all the primes below two million."""
from primes import is_prime
from time import time
import unittest

def problem010(argument = 2000000):
    """Simple summation of matching numbers."""
    return sum([i for i in xrange(2, argument) if is_prime(i)])

class TestProblem010(unittest.TestCase):
    """Verify that problem #10 is correct and timely."""
    # pylint: disable=W0201
    def runTest(self):
        """The test itself."""
        self.start = time()
        self.assertEqual(problem010(), 142913828922, "Problem #10 is wrong.")
        self.stop = time()
        self.assertLess(self.stop - self.start, 60,
            """Problem #10 takes too long.""")

class TestCaseProblem010(unittest.TestCase):
    """Verify that problem #10 test case is correct."""
    # pylint: disable=W0201
    def runTest(self):
        """The test itself."""
        self.assertEqual(problem010(10), 17, "Problem #10 test case is wrong.")

if __name__ == "__main__":
    start = time()
    answer = problem010()
    print 10, answer, time() - start
